server/src/test/java/io/spine/server/entity/DefaultStateRegistryShould.java

```diff
@@ -24,22 +24,94 @@
 import org.junit.Before;
 import org.junit.Test;
 
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import static com.google.common.collect.Maps.newConcurrentMap;
+import static io.spine.Identifier.newUuid;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
 /**
  * @author Alexander Yevsyukov
+ * @author Dmitry Ganzha
  */
 public class DefaultStateRegistryShould {
 
+    private static final String DEFAULT_STATES_FIELD_NAME = "defaultStates";
     private DefaultStateRegistry registry;
+    private Map<Object, Object> spyMap;
 
     @Before
     public void setUp() {
-        registry = new DefaultStateRegistry();
+        spyMap = spy(newConcurrentMap());
+        registry = DefaultStateRegistry.getInstance();
+        try {
+            final Field defaultStates = registry.getClass()
+                                          .getDeclaredField(DEFAULT_STATES_FIELD_NAME);
+            defaultStates.setAccessible(true);
+            defaultStates.set(registry, spyMap);
+        } catch (NoSuchFieldException | IllegalAccessException ignored) {
+        }
+    }
+
+    @Test
+    public void verify_put_invoked_one_time_when_invoke_get_default_state_in_multithreaded_environment() {
+        final int numberOfEntities = 1000;
+        final Collection<Callable<Object>> tasks = new ArrayList<>();
```

Comment:
I think that a better way to express your intention would be to use Guava's API:
```java
... = Lists.newArrayListWithExpectedSize(numberOfEntities);
``` 
(and use static import for this method).

Also, maybe a collection of `Callable<Void>` would look better. I'm not sure on the latter though.